main.py

```diff
@@ -409,93 +378,45 @@ async def delete(self) -> None:
-    """Strips and replaces &nbsp; with a space."""
-    return username.strip().replace(u"\xa0", " ")
-
-
-def parse_leader(metric: Metric, row: t.Tuple[str, ...]) -> MetricLeader:
-    """Parses a metric leader out of a tuple of strings holding the necessary
-    data points.
-    """
-    if metric.category:
-        # This is an activity or boss
-        rank, username, score = row
-        return NonSkillLeader(
-            metric,
-            clean_username(username),
-            int(clean_table_data(rank)),
-            int(clean_table_data(score)),
-        )
-    else:
-        # This is a skill
-        rank, username, level, exp = row
-        return SkillLeader(
-            metric,
-            clean_username(username),
-            int(clean_table_data(rank)),
-            int(clean_table_data(level)),
-            int(clean_table_data(exp)),
-        )
-
-
-def parse_leaders(metric: Metric, data: Tag) -> t.List[MetricLeader]:
-    """Transforms the parsed html table data into a list of metric leaders."""
-    # Skills have 4 columns, bosses and activities have 3
-    columns = 3 if metric.category else 4
-
-    # Remove empty columns (bad data at the beginning of each table)
-    # and group the necessary number of columns based on metric
-    rows = zip(*[iter(row.text for row in data if row.text)] * columns)
-
-    # Parse and return the metric leaders
-    return [parse_leader(metric, row) for row in rows]
-
-
-def build_url(metric: Metric, page: int) -> str:
+def build_url(metric: Metric, rank: int) -> str:
     """Builds the URL to use to fetch leaders for the given metric."""
-    params = {"category_type": metric.category, "table": metric.table, "page": page}
+    params = {"category": metric.category, "table": metric.table, "size": PLAYERS_PER_PAGE, "toprank": rank}
     query = "&".join(f"{k}={v}" for k, v in params.items() if v is not None)
     mode = "hiscore_oldschool"
 
     if ENABLE_SEASONAL:
         mode += "_seasonal"
 
-    return f"{BASE_URL}/m={mode}/overall?{query}"
-
+    return f"{BASE_URL}/m={mode}/ranking.json?{query}"
 
 #########################################################
 # END Utilities
 #########################################################
 
+async def fetch_hiscore_players(session: ClientSession, metric: Metric, rank: int = 0) -> t.List[HiscorePlayer]:
+    url = build_url(metric, rank)
 
-async def fetch_leaders(session: ClientSession, metric: Metric, page: int = 1) -> t.List[MetricLeader]:
-    url = build_url(metric, page)
-
-    # Fetch the hiscores data for this metric
     response = await session.get(url)
-    text = await response.text()
-
-    # Parse the HTML down to the table data
-    soup = BeautifulSoup(text, "html.parser")
-    table = t.cast(Tag, soup.findChild("table"))
-    rows = t.cast(Tag, table.findChildren("td"))
-
-    # Parse and return the metric leaders
-    return parse_leaders(metric, rows)
+    if response.status != 200:
+        return
```

Comment:
This method is supposed to return a list but here we are returning nothing which could raise an exception later (it also doesn't properly type check). we should probably log if the request fails too.